kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/IterableEq.kt

```diff
@@ -3,86 +3,163 @@ package io.kotest.assertions.eq
 import io.kotest.assertions.Actual
 import io.kotest.assertions.Expected
 import io.kotest.assertions.failure
+import io.kotest.assertions.print.Printed
 import io.kotest.assertions.print.print
 
 object IterableEq : Eq<Iterable<*>> {
 
-   /**
-    * Returns true if this [Iterable] is an iterable that is supported by this typeclass.
-    *
-    * There are many third party libs that use Iterable as a base type (Jackson's Json Node, JDK's Path type, to
-    * name but two) and we cannot safely compare these simply using the IterableEq.
-    *
-    * Therefore only explicit types are accepted by our IterableEq.
-    */
    fun isValidIterable(it: Any): Boolean {
       return when (it) {
-         is List<*>, is Set<*>, is Array<*>, is Collection<*> -> true
+         is String -> false
+         is List<*>, is Set<*>, is Array<*>, is Collection<*>, is Iterable<*> -> true
          else -> false
       }
    }
 
    fun asIterable(it: Any): Iterable<*> {
+      check(it !is String)
       return when (it) {
          is Array<*> -> it.asList()
          is List<*> -> it
          is Set<*> -> it
          is Collection<*> -> it
+         is Iterable<*> -> it
          else -> error("Cannot convert $it to Iterable<*>")
       }
    }
 
+   fun isOrderedSet(item: Iterable<*>): Boolean = item is LinkedHashSet || (item is Set<*> && item.size == 1)
```

Comment:
I did not understand the reason of checking length is 1.

Reply:
A Set of one by default is of a unique type different than `LinkedHashSet`; discussion below addresses a larger conundrum in regard to this subject.  One can also argue that a Set of 1 is ordered, but I'm picking nits here.  I'm open to any sensible disposition.